cli/azd/pkg/auth/manager.go

```diff
@@ -720,6 +720,26 @@ func (m *Manager) getSignedInAccount(ctx context.Context) (*public.Account, erro
 	return nil, nil
 }
 
+// GetLoggedInServicePrincipalClientID fetches the client ID for the signed in service principal,
+// or nil if one does not exist.
+func (m *Manager) GetLoggedInServicePrincipalClientID(ctx context.Context) (*string, error) {
```

Comment:
azd supports logging in with:
- Using `client-id`
  - service principal (secret pass)
  - service principal (certificate)
  - Federated Token
- External auth
  - using az cli as token provider
- Using web browser
  - device-code
  - Interactive auth

I wouldn't like the idea of adding a new function for each supported scenario like `GetLoggedIn<scenario-name>(ctx)`

My recommendation is:
- Keep `getSignedInAccount()` private
- Create a new structure in `auth package` (auth/manager.go) like `type LogInDetails struct`
- Define a `loginType string` and `account string` inside `LogInDetails` (to start with)
- Adda new method for the `Manager` in auth package like `LogInDetails(ctx) (*LogInDetails, error) {}` and implement it for returning the information for the actual log in.
  - Create string enum for the login type.
- Use the `LogInDetails` results from the `auth login` command.  

This approach would allow future iterations to easily add more information about the login (for example, adding the date of logged in, if device-code was used, etc.).  We would also let the auth-Manager to be the one who knows how to recover/pull the login info details, w/o the azd-command trying to pull each type to guess the one that was used.

@ellismg for reviewing the proposal.

Reply:
> I wouldn't like the idea of adding a new function for each supported scenario like GetLoggedIn<scenario-name>(ctx)

not sure if this is the case that we have here.

The current function `getSignedInAccount()` works with any type of login the problem that made me create another one is that `getSignedInAccount()` returns nil if you are signing in with a service principal account so If you are signing in with a normal email whatever sign-in method you are using `getSignedInAccount()` will work as it has been working.
That's why I created the other function to get the service principal without changing the current functionality of `getSignedInAccount()` not because it's a different scenario of login.

Reply:
My proposal's main intention is having a design which allows adding more information about the login in the future, like the type, time, etc.

It is not 100% required to complete the requirements from the associated issue, in this case. But I would take the change to set the foundation that enables getting all details about the login.

Reply:
I understand, let's wait for Matt and see.

Reply:
@rajeshkamal5050 we need Matt's input on it, what approach should we take in resolving this?

@vhvb1989 suggested a plan for it and I'm happy to implement but was waiting for any feedback from Matt to avoid rework.